apps/reference/docs/guides/auth/auth-security.mdx

```diff
@@ -0,0 +1,503 @@
+through enterprise-level identity providers via the SAML or OIDC protocols,
+provided that no account linking has taken place on those accounts with social
+login providers or with password and passwordless authentication.
+
+If your application has high privacy requirements, your
+users should obfuscate their email address by using virtual email addresses (e.g., `user+obfuscation@example.com` if allowed by their email address
+provider) or use a trusted social login provider.
+
+## Access and refresh tokens
+
+A successful user login triggers Supabase Auth to issue an access and refresh
+token.
+
+Access tokens are in the form of a JSON Web Token (JWT) and include most
+standard JWT claims for this purpose. At this time each JWT is signed using the
+`HS256` algorithm (HMAC with SHA256). The secret key used to derive the
+signature is generated from a cryptographically-secure pseudo random number
+generator when a project is initially created and is 512 bits long. This key is
+stored encrypted in our database, so Supabase staff are unable to readily
+access it.
+
+We believe in giving project owners maximum flexibility, so this key is visible
+to you only in your project's dashboard. Make sure you never share this value
+with parties you don't completely trust.
+
+:::tip
+You can always rotate the JWT signing key if you suspect it has leaked.
+Rotating keys on a regular basis is a good practice. Note that rotating a key
+means that all access tokens will immediately become invalid. Make sure you do
+not forget to update all other services that you use Supabase Auth access
+tokens in with the rotated secret to avoid any disruptions.
+:::
+
+Access tokens are also short lived, and you can configure their validity
+duration in the project's settings page. The default value at this time is 1
+hour, which is common across the industry. We do not recommend exceeding this
+duration period, and recommend using a shorter value if it makes sense for your
+application. Durations less than 5 minutes are discouraged as computers often
+may have significant clock skew that could introduce confusing and difficult to
+debug application errors and behaviors. Shortening the duration also increases
+the application's dependence of the uptime of the authentication server.
+
+The purpose of access tokens is to be readily shared between components of your
+application, be it the browser, Supabase services such as PostgREST and
+functions, your hosting or other API servers under your control. Due to
+their short lifetime, any security incident involving them will have a
+time-limited blast radius.
+
+Always make sure that access tokens are passed as headers or cookies, and avoid
+passing them as URL query parameters (regardless of request methods) to prevent
+them from appearing in access logs which may not be under your total control.
+When using them in Supabase functions or other API servers, take care to not
+accidentally write them to logs.
```

Comment:
```suggestion
Always pass access tokens as headers or cookies rather than
URL query parameters (regardless of request methods) to prevent
them from appearing in access logs.
When using access tokens in Edge Functions or other API servers, take care to not
accidentally write them to logs.
```